scripts/run_mypy_checks.py

```diff
@@ -0,0 +1,793 @@
+    'scripts/linters/general_purpose_linter_test.py',
+    'scripts/linters/html_linter.py',
+    'scripts/linters/html_linter_test.py',
+    'scripts/linters/js_ts_linter.py',
+    'scripts/linters/js_ts_linter_test.py',
+    'scripts/linters/linter_utils.py',
+    'scripts/linters/linter_utils_test.py',
+    'scripts/linters/other_files_linter.py',
+    'scripts/linters/other_files_linter_test.py',
+    'scripts/linters/pre_commit_linter.py',
+    'scripts/linters/pre_commit_linter_test.py',
+    'scripts/linters/pylint_extensions.py',
+    'scripts/linters/pylint_extensions_test.py',
+    'scripts/linters/python_linter.py',
+    'scripts/linters/python_linter_test.py',
+    'scripts/linters/warranted_angular_security_bypasses.py',
+    'scripts/pre_commit_hook.py',
+    'scripts/pre_commit_hook_test.py',
+    'scripts/pre_push_hook.py',
+    'scripts/pre_push_hook_test.py',
+    'scripts/regenerate_requirements.py',
+    'scripts/regenerate_requirements_test.py',
+    'scripts/release_scripts/cut_release_or_hotfix_branch.py',
+    'scripts/release_scripts/cut_release_or_hotfix_branch_test.py',
+    'scripts/release_scripts/repo_specific_changes_fetcher.py',
+    'scripts/release_scripts/repo_specific_changes_fetcher_test.py',
+    'scripts/release_scripts/update_changelog_and_credits.py',
+    'scripts/release_scripts/update_changelog_and_credits_test.py',
+    'scripts/release_scripts/update_configs.py',
+    'scripts/release_scripts/update_configs_test.py',
+    'scripts/run_backend_tests.py',
+    'scripts/run_custom_eslint_tests.py',
+    'scripts/run_e2e_tests.py',
+    'scripts/run_e2e_tests_test.py',
+    'scripts/run_frontend_tests.py',
+    'scripts/run_lighthouse_tests.py',
+    'scripts/run_mypy_checks.py',
+    'scripts/run_mypy_checks_test.py',
+    'scripts/run_portserver.py',
+    'scripts/run_presubmit_checks.py',
+    'scripts/run_tests.py',
+    'scripts/script_import_test.py',
+    'scripts/scripts_test_utils.py',
+    'scripts/scripts_test_utils_test.py',
+    'scripts/servers.py',
+    'scripts/servers_test.py',
+    'scripts/setup.py',
+    'scripts/setup_gae.py',
+    'scripts/setup_gae_test.py',
+    'scripts/setup_test.py',
+    'scripts/start.py',
+    'scripts/third_party_size_check.py',
+    'scripts/typescript_checks.py',
+    'scripts/typescript_checks_test.py',
+]
+
+
+CONFIG_FILE_PATH = os.path.join('.', 'mypy.ini')
+# TODO(#13113): Change mypy command to mypy path after Python3 migration.
+MYPY_CMD = 'mypy'
+MYPY_REQUIREMENTS_PATH = os.path.join('.', 'mypy_requirements.txt')
+PYTHON3_CMD = 'python3'
+
+
+_PARSER = argparse.ArgumentParser(
+    description='Type checking script for Oppia codebase.'
+)
+
+_PARSER.add_argument(
+    '--files',
+    help='Files to type-check',
+    action='append',
+    nargs='+'
+)
+
+
+def get_mypy_cmd(files):
+    """Return the appropriate command to be run."""
+    if files:
+        cmd = [MYPY_CMD, '--config-file', CONFIG_FILE_PATH] + files[0]
```

Comment:
This looks strange in the first look. The behaviour you assume is for `extend` argparse action. But that is only available in python3.8+. So I used `append` argparse action. It gave `files = [['file1', file2]]`. Thanks!

Reply:
I would suggest only passing `['file1', file2]` to this `get_mypy_cmd` and have a clear docstring, will that e fine?

Also, I think if you'll change action for this arg to `store` instead of `append` then it will fix the issue.

Reply:
Thanks, it worked.

Reply:
Changed